base/shell/cmd/cmd.c

```diff
@@ -2333,28 +2333,11 @@ static VOID Cleanup(VOID)
         ParseCommandLine(_T("\\cmdexit.bat"));
     }
 
-#ifdef FEATURE_DIRECTORY_STACK
-    /* Destroy directory stack */
-    DestroyDirectoryStack();
-#endif
-
-#ifdef FEATURE_HISTORY
-    CleanHistory();
-#endif
-
-    /* Free GetEnvVar's buffer */
-    GetEnvVar(NULL);
-
-    /* Remove ctrl break handler */
-    RemoveBreakHandler();
-
     /* Restore the default console mode */
     SetConsoleMode(ConStreamGetOSHandle(StdIn),
                    ENABLE_PROCESSED_INPUT | ENABLE_LINE_INPUT | ENABLE_ECHO_INPUT);
     SetConsoleMode(ConStreamGetOSHandle(StdOut),
                    ENABLE_PROCESSED_OUTPUT | ENABLE_WRAP_AT_EOL_OUTPUT);
-
-    DeleteCriticalSection(&ChildProcessRunningLock);
```

Comment:
Could this interfere with application verifier?
There are some stops for critical sections not being released, but I am not sure if that also  is for the main thread.

Reply:
As far as I know, application verifier checks that you properly call 'DeleteCriticalSection' before freeing its underlying memory. In this case, this would be when Mm frees all pages from the mapped executable, when application verifier is way out of play.